### DISCUSSION TITLE
Make mA an officially supported unit of current

### DISCUSSION THREAD

**Discussion by SukramJ** [home-assistant/architecture#847] [Entity Models] [answered]
Currently some integration (wled, tuya, homematic, issy994) use `mA` as a `unit of current`, but this has never been part of the official developer documentation.

I suggest to officially support mA as unit of current, as this is a usual fraction of the already supported unit `A`.

Not adding this will cause a lot of warnings in users logs, due to the SensorDeviceClass to unit check to be introduced with HA 2023.01.
`... is using native unit of measurement 'mA' which is not a valid unit for the device class ('current') it is using; Please update your configuration if your entity is manually configured, ...`

Proposed changes can be seen in these PR:
core: https://github.com/home-assistant/core/pull/84492
doc: https://github.com/home-assistant/home-assistant.io/pull/25408
dev-doc: https://github.com/home-assistant/developers.home-assistant/pull/1593
frontend: https://github.com/home-assistant/frontend/pull/14916

**Comment by frenck** [comment]
It looks like the `current` device class (with support for just `A` was introduced in:

https://github.com/home-assistant/architecture/issues/397

I don't see a direct reason in that original discussion why it was limited to A.

**Comment by frenck** [comment]
Discussed a bit with @MartinHjelmare, and we think we should ensure this also adds conversion capabilities. So the user can change the unit in the UI.

Example for data size:
- https://github.com/home-assistant/core/pull/84699
- https://github.com/home-assistant/frontend/pull/14903

**Reply by SukramJ** [reply]
I think the expected `NORMALIZED_UNIT` is `A`?

**Reply by SukramJ** [reply]
I added conversion capabilities to frontend & backend.

**Comment by semihozsoy** [comment]
Guys sorry to interrupt but I do not know who is add but you add the wrong guy on the project. I am not the one who is working with you. I think there is misunderstanding of email 😄
________________________________
Gönderen: SukramJ ***@***.***>
Gönderildi: 29 Aralık 2022 Perşembe 12:13
Kime: home-assistant/architecture ***@***.***>
Bilgi: Semih Özsoy ***@***.***>; Mention ***@***.***>
Konu: Re: [home-assistant/architecture] Make mA an officially supported unit of current (Discussion #847)


I think the expected NORMALIZED_UNIT is A?

**Reply by frenck** [reply]
It was spam, it shared a link to a third-party website, mentioning tons of users in the message. 
I deleted the message, banned the user, and reported it with GitHub.

**Comment by abhishek-dangol** [comment]
All,

I think this whole thing is one giant mess!

 Let’s scratch everything that has been done so far and start from the very
beginning.

The last thing I want is to deliver a product that does not meet even the
basic requirements and by the looks of it this train wreck is heading in
that direction!

Speak up if any of you do not concur!

On Thu, Dec 29, 2022 at 16:46 SukramJ ***@***.***> wrote:

> I added conversion capabilities to frontend & backend.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/home-assistant/architecture/discussions/847#discussioncomment-4552672>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AMKUMYGTIDVAMDXFZTDK2D3WPYIDNANCNFSM6AAAAAATKHYE6M>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***
> com>
>
-- 

Regards,

Abhishek "Dan" Dangol

***@***.***

832-790-5551 <8327905551>

Houston, Texas, USA

<https://www.linkedin.com/in/abhishekdangol/>

<https://github.com/abhishek-dangol>

**Comment by frenck** [comment, marked as answer]
I've been discussing this in multiple places on Discord with multiple (groups of) people. We all seem to agree that this addition is good to go.

Considering this popping up in the current beta (due to the new checks/warnings), we are even going to pull it into the current beta to reduce the number of reports, and to reduce the number of integration removing the device class because of it.

Not pulling it into beta now, would mean people would remove the device classes that later get supported again.

I'll pick up the PRs linked above to ensure to land into the 2023.1 beta.

Thanks for the proposal @SukramJ 👍 

../Frenck

**Comment by lipzh-yzr** [comment]
邮件已收到。
我会尽快回复。
Your mail has been received and I will reply soon.
Best wishes to you.
